plugins/content/courseasset/index.js

```diff
@@ -42,38 +42,67 @@ CourseAssetContent.prototype.getModelName = function () {
 function initialize() {
   // Content Hook for updatedAt and updatedBy:
   ['course', 'contentobject', 'article', 'block', 'component'].forEach(function (contentType) {
-    app.contentmanager.addContentHook('destroy', contentType, { when: 'pre' }, function (contentType, data, next) { 
-
+    app.contentmanager.addContentHook('destroy', contentType, { when: 'pre' }, function (contentType, data, next) {
       database.getDatabase(function (err, db) {
         db.retrieve(contentType, data[0], function(err, items) {
           if (err) {
             logger.log('err', err);
             return next(err);
           }
-
           if (items && items.length == 1) {
             var itemForDeletion = items[0].toObject();
             var criteria = {};
-            
+
             async.series([
               function(callback){
+                if (contentType !== 'contentobject') {
+                  return callback(null, 'Not processing a content object');
+                }
+                db.retrieve('article', { _courseId: itemForDeletion._courseId, _parentId: itemForDeletion._id }, function(err, articles) {
+                  if (err) {
+                    return callback(err, 'Unable to retrieve child articles of contentobject ' + itemForDeletion._id);
+                  }
+                  var parentIds = [];
+                  if (articles && articles.length !== 0) {
```

Comment:
@taylortom, I think there's an issue here if you attempt to delete a page which has no articles.  It's an edge case, I know.  Can you add something like the following instead?
```
                  if (articles.length == 0) {
                    return callback(null, 'A contentobject with no articles is being removed');
                  } 

```